pycqed/measurement/hdf5_data.py

```diff
@@ -124,9 +125,135 @@ def encode_to_utf8(s):
+        if isinstance(item, (str, tuple, float, int, bool,
+                             np.float_, np.int_, np.bool_)):
+            entry_point.attrs[key] = item
+        elif isinstance(item, np.ndarray):
+            entry_point.create_dataset(key, data=item)
+        elif item is None:
+            # as h5py does not support saving None as attribute
+            # I create special string, note that this can create
+            # unexpected behaviour if someone saves a string with this name
+            entry_point.attrs[key] = 'NoneType:__None__'
+
+        elif isinstance(item, dict):
+            entry_point.create_group(key)
+            write_dict_to_hdf5(data_dict=item,
+                               entry_point=entry_point[key])
+        elif isinstance(item, list):
+            if len(item) > 0:
+                elt_type = type(item[0])
+                # Lists of a single type, are stored as an hdf5 dset
+                if (all(isinstance(x, elt_type) for x in item) and
+                        not isinstance(item[0], dict)):
+                    if isinstance(item[0], (int, float,
+                                            np.int32, np.int64)):
+                        entry_point.create_dataset(key,
+                                                   data=np.array(item))
+                        entry_point[key].attrs['list_type'] = 'array'
+
+                    # strings are saved as a special dtype hdf5 dataset
+                    elif isinstance(item[0], str):
+                        dt = h5py.special_dtype(vlen=str)
+                        data = np.array(item)
+                        data = data.reshape((-1, 1))
+                        ds = entry_point.create_dataset(
+                            key, (len(data), 1), dtype=dt)
+                        ds.attrs['list_type'] = 'str'
+                        ds[:] = data
+                    else:
+                        logging.warning(
+                            'List of type "{}" for "{}":"{}" not '
+                            'supported, storing as string'.format(
+                                elt_type, key, item))
+                        entry_point.attrs[key] = str(item)
+                # Storing of generic lists
+                else:
+                    entry_point.create_group(key)
+                    # N.B. item is of type list
+                    list_dct = {'list_idx_{}'.format(idx): entry for
+                                idx, entry in enumerate(item)}
+                    group_attrs = entry_point[key].attrs
+                    group_attrs['list_type'] = 'generic_list'
+                    group_attrs['list_length'] = len(item)
+                    write_dict_to_hdf5(
+                        data_dict=list_dct,
+                        entry_point=entry_point[key])
+            else:
+                # as h5py does not support saving None as attribute
+                entry_point.attrs[key] = 'NoneType:__emptylist__'
+
+        else:
```

Comment:
don't forget to check tuples:
elif isinstance(item, tuple):
//now you can do the same as with lists